www/content/workstation/upgrade_labs.md

```diff
@@ -0,0 +1,470 @@
+
+```
+------------------------ WARNING ---------------------------
+Changes made to the following cookbooks in ./node-MYNODE-repo/cookbooks
+cannot be saved to the cookbook's source, though they can still be uploaded
+to a Chef Server:
+
+  - logrotate (v1.9.2)
+  - windows (v1.44.1)
+  - chef_handler (v1.4.0)
+-----------------------------------------------------------
+
+You're ready to begin!
+
+Start with 'kitchen converge'.  As you identify issues, you
+can modify cookbooks in their original checkout locations or
+in the repository's cookbooks directory and they will be picked
+up on subsequent runs of 'kitchen converge'.
+```
+
+## Detect and Correct Cookbook Errors
+
+{{< note >}}
+Be cautious when running your cookbooks locally. It is possible to accidentally impact production infrastructure based on settings in your cookbooks and attributes. Read over your cookbooks and attributes, especially attributes set in roles and environments.  If needed, override attributes to be appropriate for local testing in your `kitchen.yml`.
+{{< /note >}}
+
+### Increment the Chef Infra Client Version
+
+In the `kitchen.yml` file, change the `product_version` line to `16`:
+
+```yaml
+provisioner:
+  name: chef_zero
+  product_name: chef
+  product_version: 16  # Change this line
+```
+
+If needed, you can "step forward" by first going from 12 to 13, correcting issues, then 13 to 14, etc.
+
+### Attempt a Converge and Check for Errors
+
+Save the file and test it in the `node-MY_NODE-repo` directory by running:
+
+```
+chef exec kitchen converge
+```
+
+Watch for Chef Infra errors. If any occur, fix them.
+
+### Test and Correct with Cookstyle
+
+To check for version upgrade issues, run:
+
+```
+$ chef exec cookstyle cookbooks/my_cookbook
+```
+
+Repeat this process for each cookbook for that node.
+
+#### Using Cookstyle Auto-correct
+
+Use Cookstyle's auto-correct feature to fix style errors by adding the `-a` (for auto-correct) flag:
+
+```
+$ chef exec cookstyle -a cookbooks/my_cookbook
+```
+
+Other issues may require manual intervention and editing.
+Repeat this process for each cookbook that the node consumes.
+
+### Copy Data Bags
+
+If data bags are used on your Chef Infra Server, you will need to download the `data_bags` directory in your repository.
+Note that this command does not support embedded keys in credentials files. If you use embedded keys, move the key to a key file.
+
+```
+cd node-node-01-repo
+chef exec knife download data_bags --chef-repo-path . --profile old-server --key my-old-key.pem
+```
+
+## Deploy your Chef Lab Upgrades
+
+{{< note >}}
+This guide suggests migrating upgraded cookbooks and nodes to a new Chef Server. This pattern is not feasable for all customers, specifically ones who rely on Chef search for inventory and coordination. But we feel migrating to a new server works for customers who do not rely on search because it creates a fresh start to build on for future migration to Effortless.
```

Comment:
```suggestion
This guide suggests migrating upgraded cookbooks and nodes to a new Chef Server. This pattern is not feasible for all customers, specifically ones who rely on Chef search for inventory and coordination. But we feel migrating to a new server works for customers who do not rely on search because it creates a fresh start to build on for future migration to Effortless.
```

Reply:
What prevents the migration to a newer server? Is it a question of the Chef Infra Server version or of existing search indexes?

Reply:
How about:
```
This guide advises migrating upgraded cookbooks and nodes to a new Chef Infra Server. Migrating to a new Chef Infra Server is generally a good solution for most customers because it creates a fresh start to build on for future migration to Effortless. However, we specifically advise _against_ upgrading the Chef Infra Server if your organization relies on Chef search for inventory and coordination. 
```